roofit/roofitcore/inc/RooAbsReal.h

```diff
@@ -109,7 +109,18 @@ class RooAbsReal : public RooAbsArg {
 
   /// \deprecated getValBatch() has been removed in favour of the faster getValues(). If your code is affected
   /// by this change, please consult the release notes for ROOT 6.24 for guidance on how to make this transition.
-  virtual RooSpan<const double> getValBatch(std::size_t begin, std::size_t maxSize, const RooArgSet* normSet = nullptr) = delete;
+  virtual RooSpan<const double> getValBatch(std::size_t /*begin*/, std::size_t /*maxSize*/, const RooArgSet* /*normSet*/ = nullptr)
+#ifndef R__MACOSX
+  = delete;
+#else
+  //AppleClang in MacOS10.14 has a linker bug and fails to link programs that create objects of classes containing virtual deleted methods.
+  //This can be safely deleted when MacOS10.14 is no longer supported by ROOT. See https://reviews.llvm.org/D37830
+  final {
+    throw std::runtime_error("Deprecated getValBatch() has been removed in favour of the faster getValues(). If your code is affected by this change, please consult the release notes for ROOT 6.24 for guidance on how to make this transition.");
+  }
+#endif
+
```

Comment:
I suspect that this will throw off the doxygen parser, and therefore create broken docs. I suggest:
```suggestion
#ifndef R__MACOSX
  virtual RooSpan<const double> getValBatch(std::size_t /*begin*/, std::size_t /*maxSize*/, const RooArgSet* /*normSet*/ = nullptr) = delete;
#else
  //AppleClang in MacOS10.14 has a linker bug and fails to link programs that create objects of classes containing virtual deleted methods.
  //This can be safely deleted when MacOS10.14 is no longer supported by ROOT. See https://reviews.llvm.org/D37830
  virtual RooSpan<const double> getValBatch(std::size_t /*begin*/, std::size_t /*maxSize*/, const RooArgSet* /*normSet*/ = nullptr) final {
    throw std::runtime_error("Deprecated getValBatch() has been removed in favour of the faster getValues(). If your code is affected by this change, please consult the release notes for ROOT 6.24 for guidance on how to make this transition.");
  }
#endif
```

Reply:
Similarly for the other function.

Also:
Suggest to use `std::logic_error`, as it indicates faulty logic in the program.
Runtime error is meant for stuff that's beyond what the program can control.